selfdrive/controls/monitord.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+import gc
+from common.realtime import sec_since_boot, set_realtime_priority
+from common.params import Params, put_nonblocking
+import cereal.messaging as messaging
+from selfdrive.controls.lib.drive_helpers import create_event, EventTypes as ET
+from selfdrive.controls.lib.driver_monitor import DriverStatus, MAX_TERMINAL_ALERTS, MAX_TERMINAL_DURATION
+from selfdrive.locationd.calibration_helpers import Calibration
+from selfdrive.controls.lib.gps_helpers import is_rhd_region
+
+def monitord_thread(sm=None, pm=None):
+  gc.disable()
+
+  # start the loop
+  set_realtime_priority(3)
+
+  params = Params()
+
+  # Pub/Sub Sockets
+  if pm is None:
+    pm = messaging.PubMaster(['monitorState'])
+
+  if sm is None:
+    sm = messaging.SubMaster(['driverMonitoring', 'liveCalibration', 'carState', 'model', 'gpsLocation'], ignore_alive=['gpsLocation'])
+
+  driver_status = DriverStatus()
+  is_rhd = params.get("IsRHD")
+  if is_rhd is not None:
+    driver_status.is_rhd = bool(int(is_rhd))
+
+  sm['liveCalibration'].calStatus = Calibration.INVALID
+  v_cruise_last = 0
+
+  # wait until carState live
+  CS_updated = False
+  while(not CS_updated):
+    sm.update(0)
+    CS_updated = sm.updated['carState']
+
+  # 10Hz <- monitoringd
+  while(True):
+    start_time = sec_since_boot()
+    sm.update(0)
```

Comment:
Don't update with zero timeout. Use the default timeout if there is no specific reason to change it.